evalml/data_checks/datetime_format_data_check.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+
+from evalml.data_checks import DataCheck, DataCheckError, DataCheckMessageCode
+from evalml.utils import infer_feature_types
+
+
+class DateTimeFormatDataCheck(DataCheck):
+    """Checks if the datetime column has equally spaced intervals and is monotonically increasing or decreasing in order
+    to be supported by time series estimators.
+
+    Arguments:
+        datetime_column (str): The name of the datetime column. If the datetime values are in the index, then pass "index".
+
+    """
+
+    def __init__(self, datetime_column="index"):
+        self.datetime_column = datetime_column
+
+    def validate(self, X, y):
+        """Checks if the target data has equal intervals and is sorted.
+
+        Arguments:
+            X (pd.DataFrame, np.ndarray): Features.
+            y (pd.Series, np.ndarray): Target data.
+
+        Returns:
+            dict (DataCheckError): List with DataCheckErrors if unequal intervals are found in the datetime column.
+
+        Example:
+            >>> from pandas as pd
+            >>> X = pd.DataFrame(pd.date_range("January 1, 2021", periods=8), columns=["dates"])
+            >>> y = pd.Series([1, 2, 4, 2, 1, 2, 3, 1])
+            >>> X.iloc[7] = "January 9, 2021"
+            >>> datetime_format_check = DateTimeFormatDataCheck()
+            >>> assert datetime_format_check.validate(X, y) == {"errors": [{"message": "No frequency could be detected in dates, possibly due to uneven intervals.",\
+                                                                    "data_check_name": "EqualIntervalDataCheck",\
+                                                                    "level": "error",\
+                                                                    "code": "DATETIME_HAS_UNEVEN_INTERVALS",\
+                                                                    "details": {}}],\
+                                                        "warnings": [],\
+                                                        "actions": []}
+        """
+        results = {"warnings": [], "errors": [], "actions": []}
+
+        X = infer_feature_types(X)
+        y = infer_feature_types(y)
+
+        if self.datetime_column != "index":
+            datetime_values = X[self.datetime_column]
+        else:
+            datetime_values = X.index
+            if not isinstance(datetime_values, pd.DatetimeIndex):
+                datetime_values = y.index
+            if not isinstance(datetime_values, pd.DatetimeIndex):
+                raise TypeError(
+                    "Either X or y has to have datetime information in its index."
+                )
+
+        try:
+            inferred_freq = pd.infer_freq(datetime_values)
+        except TypeError:
+            raise TypeError(
```

Comment:
I think rather than reraising or raising the type error we should just return a DataCheckError saying that the column cannot be treated as a date.

I think that fits in better with our data check pattern of returning instances of DataCheckErrors/Warnings rather than raising exceptions.